src/map.cpp

```diff
@@ -6804,6 +6804,18 @@ void map::restock_fruits( const tripoint &p, const time_duration &time_since_las
     }
 }
 
+void map::restore_cut_plant( const tripoint &p, const time_duration &time_since_last_actualize )
+{
+    const auto &ter = this->ter( p ).obj();
+    if( !ter.has_flag( TFLAG_CUT ) ) {
+        return; // Plant isn't cut.  Do nothing.
+    }
+    // Restore the tree to normal growth so that eventually it can bear fruit.
+    if( time_since_last_actualize >= calendar::year_length() ) {
```

Comment:
Pretty sure that isn't what you want.
`time_since_last_actualize` depends on the time since you last loaded that map segment. So this will only trigger if you never load this map segment for a whole year. If you visit the tree every 90% of a year, for 10 years, it will not trigger.

Actually noting the date of when the cut was made is much harder than it sounds, so you're better off just making the cuttings always kill the tree and produce multiple cuttings at a time.